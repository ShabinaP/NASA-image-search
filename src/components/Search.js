import React, { useState } from "react";
import "../styles/Search.css"
import "../requests/getImages"
import getImages from "../requests/getImages";

const Search = ({setSearchResults}) => {
const [value, setValue] = useState("")
const handleSubmit = (event) => {
    event.preventDefault()
    setSearchResults(getImages(value))
}
    return (
        <> 
        <form  className="search_form" onSubmit={handleSubmit}>
        <input className="search_input" type="text" placeholder="Search here.." onChange={(e) => setValue(e.target.value)} /> 
        <button className="submit_button" type="submit"> Shoot! </button>
        </form> </>
    )

    }

    export default Search;