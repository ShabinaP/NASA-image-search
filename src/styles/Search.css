.search_input {
    margin: auto;
    width: 25%;
    height: 40px;
    border-radius: 11px;
    font-size: 20px;
    padding: 0 15px;
}