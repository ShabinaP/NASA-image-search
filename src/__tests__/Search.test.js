import React from "react";
import { render } from '@testing-library/react';
import Search from "../components/Search";


describe("Search", () => {
    const validProps = {
        setSearchResults: () => {}
    }
    const {asFragment} = render(<Search
    setSearchResults={validProps.setSearchResults}
    />);

    it("renders correctly", () => {
        expect(asFragment()).toMatchSnapshot()
    })
}) 