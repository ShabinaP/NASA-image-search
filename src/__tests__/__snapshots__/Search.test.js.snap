// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search_form"
  >
    <input
      class="search_input"
      placeholder="Search here.."
      type="text"
    />
    <button
      class="submit_button"
      type="submit"
    >
       Shoot! 
    </button>
  </form>
   
</DocumentFragment>
`;
